ch_pipeline/analysis/source_removal.py

```diff
@@ -484,6 +538,89 @@ def process(self, data):
         return out
 
 
+class LPFSourceAmplitude(task.SingleTask):
+    """Apply a 2D low-pass filter to the measured source amplitude.
+
+    Attributes
+    ----------
+    window : list
+        The size of the moving average window along the
+        [freq, time] axis.  This sets the cutoff scale of
+        the low-pass filter.
+    niter : list
+        Number of iterations of the moving average filter
+        along the [freq, time] axis.  The peak-to-sidelobe
+        ratio of the filter's transfer function scales
+        roughly as 0.05^niter.
+    frac_required : float
+        The fraction of samples within a window that must be
+        valid (unmasked) in order for the filtered data point
+        to be considered valid.
+    ignore_main_lobe : boolean
+        Do not apply the filter within the main lobe of the
+        primary beam.
+    main_lobe_threshold : float
+        If the source amplitude is greater than this fraction
+        of the source flux than than [freq, RA] is considered
+        within main lobe.  Only relevant when ignore_main_lobe
+        is True.
+    """
+
+    window = config.Property(proptype=list, default=[3, 5])
+    niter = config.Property(proptype=list, default=[12, 8])
+    frac_required = config.Property(proptype=float, default=0.40)
+
+    ignore_main_lobe = config.Property(proptype=bool, default=True)
+    main_lobe_threshold = config.Property(proptype=float, default=0.05)
+
+    def process(self, model):
+        """Low-pass filter the provided source model.
+
+        Parameters
+        ----------
+        model : containers.SourceModel
+            Best-fit parameters of the source model.
+
+        Returns
+        -------
+        model : containers.SourceModel
+            The input container with the amplitude dataset
+            filtered and the weights updated.
```

Comment:
May be I missed it, but where are weights being updated in this function?

Reply:
I've updated the doctoring to give a better explanation.